dist/src/main/java/io/camunda/zeebe/broker/BrokerClusterConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker;
+
+import io.atomix.cluster.AtomixCluster;
+import io.atomix.cluster.ClusterConfig;
+import io.atomix.utils.Version;
+import io.camunda.zeebe.broker.clustering.ClusterConfigFactory;
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.util.VersionUtil;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.web.context.annotation.ApplicationScope;
+
+@Configuration(proxyBeanMethods = false)
```

Comment:
:question: Why is setting this to false required?

Reply:
It's not required, but it's more best practice for our code base. If it's true, then CGLIB will try to proxy the beans so it can apply aspects. This will fail when using final classes, which we do by default. To avoid others running into this I just set it usually (plus, we don't really make use of aspects, and proxying just makes debugging more cumbersome).

Reply:
So if this is a true, a `@Bean` annotated method will not return a, say, `ActorScheduler`, but a CGLIB proxy of the `ActorScheduler`. Looks and behaves the same, but may be aspect-weaved. Downside is it must be extensible, I think (or at least it fails on final classes).